### PR TITLE
Feat: load GTM

### PR DESCRIPTION
**Pull request by katspaugh**
## What it solves
* Load the GTM module iff the cookie consent is given
* Clear the GA cookies if the consent is withdrawn
* In prod, send the tracking events to GTM
* Refactor the paths from which event types are imported
* Refactor the Beamer initialization – move out of the cookies middleware

### CODE REVIEW

File: src/hooks/wallets/useOnboard.ts [hunk 1]
```diff
@@ -46,6 +46,31 @@ export const getConnectedWallet = (wallets: WalletState[]): ConnectedWallet | nu
   }
 }
 
+const getWalletConnectLabel = async ({ label, provider }: ConnectedWallet): Promise<string | undefined> => {
+  if (label.toUpperCase() !== 'WALLETCONNECT') return
```
Comment (usame-algan) [inline src/hooks/wallets/useOnboard.ts:50]:
```suggestion
  if (label.toUpperCase() !== WALLET_KEYS.WALLETCONNECT) return
```

File: src/services/analytics/gtm.ts [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+  if (!GOOGLE_TAG_MANAGER_ID || !GTM_ENVIRONMENT.auth) {
+    console.warn('[GTM] - Unable to initialize Google Tag Manager. `id` or `gtm_auth` missing.')
+    return
+  }
+
+  TagManager.initialize({
+    gtmId: GOOGLE_TAG_MANAGER_ID,
+    ...GTM_ENVIRONMENT,
+    dataLayer: {
+      // Block JS variables and custom scripts
```
Comment (usame-algan) [inline src/services/analytics/gtm.ts:50]: Is there a reason we are not adding `event: GTM_EVENT.PAGEVIEW` etc. here anymore? If I remember correctly we used that custom event to trigger an initial page view

File: src/definitions.d.ts [hunk 3]
```diff
@@ -3,8 +3,15 @@ import { BeamerConfig, BeamerMethods } from '@services/beamer/types'
 
 declare global {
   interface Window {
+    isDesktop?: boolean
+    ethereum?: {
+      autoRefreshOnNetworkChange: boolean
+      isMetaMask: boolean
+    }
```
Comment (iamacook) [inline src/definitions.d.ts:10]: Were these ported over from `safe-react`? I know that `isDesktop` was our flag for the Electron app and the other two, I don't think we need as we never access the `window` object for them afaik.

I would say the same for `Cypress` below but I think that will be used for the injected test wallet.

File: src/services/analytics/gtm.ts (hunk 2 above)
Comment (katspaugh) [reply src/services/analytics/gtm.ts:50]: I'm planning to trigger pageViews from another place. This fn here will just intialize GTM.

File: src/hooks/useGtm.ts [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+import { CookieType, selectCookies } from '@/store/cookiesSlice'
+import useChainId from './useChainId'
+
+const useGtm = () => {
+  const chainId = useChainId()
+  const cookies = useAppSelector(selectCookies)
+  const isAnalyticsEnabled = cookies[CookieType.ANALYTICS]
+
+  useEffect(() => {
+    isAnalyticsEnabled ? gtmInit() : gtmClear()
```
Comment (iamacook) [inline src/hooks/useGtm.ts:13]: I think this should go in the `cookiesMiddleware`.

File: src/hooks/wallets/useOnboard.ts [hunk 5]
```diff
@@ -46,6 +47,31 @@ export const getConnectedWallet = (wallets: WalletState[]): ConnectedWallet | nu
   }
 }
 
+const getWalletConnectLabel = async ({ label, provider }: ConnectedWallet): Promise<string | undefined> => {
+  if (label.toUpperCase() !== WALLET_KEYS.WALLETCONNECT) return
```
Comment (iamacook) [inline src/hooks/wallets/useOnboard.ts:51]:
```suggestion
  if (label.toUpperCase() !== WALLET_KEYS.WALLETCONNECT.toUpperCase()) return
```

File: src/services/analytics/index.ts [hunk 6]
```diff
@@ -0,0 +1,15 @@
+import { AnalyticsEvent } from './types'
+import { IS_PRODUCTION } from '@/config/constants'
+import { gtmTrack } from './gtm'
+
+export const trackEvent = (eventData: AnalyticsEvent) => {
+  if (!IS_PRODUCTION) {
```
Comment (iamacook) [inline src/services/analytics/index.ts:6]: I think this is the cause of @francovenica's issues regarding no logging.

File: src/hooks/useGtm.ts (hunk 4 above)
Comment (katspaugh) [reply src/hooks/useGtm.ts:13]: I think the opposite. Cookies is just a storage, it shouldn't control what modules are loaded. I would in fact move the Beamer init out of it.

File: src/services/analytics/index.ts (hunk 6 above)
Comment (katspaugh) [reply src/services/analytics/index.ts:6]: Let's enable logging on prod for now. 👍 I'll remove the check.

File: src/hooks/useGtm.ts (hunk 4 above)
Comment (iamacook) [reply src/hooks/useGtm.ts:13]: [Middleware are intended to contain logic with side effects.](https://redux.js.org/tutorials/fundamentals/part-4-store#middleware-use-cases)

Comment (katspaugh) [reply src/hooks/useGtm.ts:13]: Related to the store, sure, but tracking and cookies are completely unrelated pieces of functionality. The only connection is that if the cookies aren't allowed, we shouldn't load the analytics. It doesn't mean the cookie storage should be responsible for loading GTM.

Comment (katspaugh) [reply src/hooks/useGtm.ts:13]: By your logic, we shouldn't use modules and write the whole app in the same file.

File: src/definitions.d.ts (hunk 3 above)
Comment (katspaugh) [reply src/definitions.d.ts:10]: I think it's for the mock injected wallet.

**Review by iamacook — APPROVED** [review]

**Pull request merged**